app/(main)/settings/team/actions.ts

```diff
@@ -44,3 +73,135 @@ export async function updateTeamName(teamDbId: number, formData: FormData) {
 		return { success: false, error };
 	}
 }
+
+export async function addTeamMember(formData: FormData) {
+	try {
+		const email = formData.get("email") as string;
+		const role = formData.get("role") as string;
+
+		if (!isTeamRole(role)) {
+			throw new Error("Invalid role");
+		}
+
+		// 1. Get current user's team
+		const supabaseUser = await getUser();
+
+		const team = await db
+			.select({ dbId: teams.dbId })
+			.from(teams)
+			.innerJoin(teamMemberships, eq(teams.dbId, teamMemberships.teamDbId))
+			.innerJoin(
+				supabaseUserMappings,
+				eq(teamMemberships.userDbId, supabaseUserMappings.userDbId),
+			)
+			.where(eq(supabaseUserMappings.supabaseUserId, supabaseUser.id));
+
+		if (team.length === 0) {
+			throw new Error("Team not found");
+		}
+
+		const currentTeam = team[0]; // TODO: This will need to be adjusted after implementing team switching
+
```

Comment:
Since Server actions must be treated as Public endpoint, please validate these, here:

- Current Team is a Pro plan (including internal) team.
- Current User is an Admin role user.

ref
https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#security

Reply:
56434bca8a6b fixed ✍️